analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AttributeDataExtensionsTest.cs

```diff
@@ -70,5 +160,30 @@ private static AttributeData AttributeDataWithName(string attributeClassName)
             attributeData.Protected().Setup<INamedTypeSymbol>("CommonAttributeClass").Returns(namedType.Object);
             return attributeData.Object;
         }
+
+        private static AttributeData AttributeDataWithArguments(IDictionary<string, object> namedArguments = null, IDictionary<string, object> constructorArguments = null)
```

Comment:
This is very heavy scaffolding, that is hard to review. What about compiling a small snippet instead and just taking the value from there?
There's so much going on that I can't verify that this is how the framework really behaves.